src/main/java/executor/service/service/ItemHandler.java

```diff
@@ -0,0 +1,7 @@
+package executor.service.service;
+
+public interface ItemHandler<T> {
+
+    void onScenarioReceived(T item);
```

Comment:
Please do not reinvent the weel!!! Java already has Consumer functional interface. You could use a Consumer instead my proposed interface, but you would lose the semantic clarity that a custom functional interface like ScenarioHandler provides. Using a custom functional interface allows you to give a meaningful name to the callback method, making your code more self-explanatory and maintainable.

Reply:
Using Consumer is a good idea. Thanks, I'll definitely use it.

Using parameterized classes or methods can be useful when you want to create generic components that can work with different data types. This can save you time and reduce code duplication, since you can create one TaskWorker implementation that works with different T types, rather than creating separate classes for each listener type.

This is a guaranteed advantage.

Here, when using the TaskWork API, you can clearly see what type is being transmitted:
    @Override
    public void execute() {
        threadPoolExecutor.execute(threadFactory.createTaskWorker(scenarioSourceListener, SCENARIO_QUEUE));

        threadPoolExecutor.execute(threadFactory.createTaskWorker(proxySourcesClient, PROXY_QUEUE));

        executeScenarioAndProxy();
    }